### PR TITLE
[VEN-2684]: Deploy IL to zksync sepolia

### PR DESCRIPTION
**Pull request by GitGuru7**
## Description

Resolves #

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: deploy/006-deploy-pool-registry.ts [hunk 1]
```diff
@@ -15,20 +15,30 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
+      viaAdminContract: {
+        name: "DefaultProxyAdmin",
+        artifact: defaultProxyAdmin,
+      },
       upgradeIndex: 0,
     },
     autoMine: true,
     log: true,
+    skipIfAlreadyDeployed: true,
+    // maxFeePerGas: "200000000"  // Needed for zksync
```
Comment (chechu) [inline deploy/006-deploy-pool-registry.ts:41]: Wouldn't we make this conditional to the network, only setting it for the zksync chains?

File: deploy/006-deploy-pool-registry.ts [hunk 2]
```diff
@@ -15,20 +15,30 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
-      proxyContract: "OpenZeppelinTransparentProxy",
+      proxyContract: "OptimizedTransparentUpgradeableProxy",
       execute: {
         methodName: "initialize",
         args: [accessControlManagerAddress],
       },
+      viaAdminContract: {
+        name: "DefaultProxyAdmin",
+        artifact: defaultProxyAdmin,
+      },
```
Comment (chechu) [inline deploy/006-deploy-pool-registry.ts:35]: Would this be compatible with the deployment to other networks? (i.e. to Optimism)

File: deploy/006-deploy-pool-registry.ts [hunk 3]
```diff
@@ -15,20 +15,30 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
   );
   const proxyOwnerAddress = await toAddress(preconfiguredAddresses.NormalTimelock || "account:deployer", hre);
 
+  const defaultProxyAdmin = await hre.artifacts.readArtifact(
+    "hardhat-deploy/solc_0.8/openzeppelin/proxy/transparent/ProxyAdmin.sol:ProxyAdmin",
+  );
+
```
Comment (chechu) [inline deploy/006-deploy-pool-registry.ts:21]: I would add a comment mentioning why we are doing this

File: deploy/010-deploy-reward-distributors.ts [hunk 4]
```diff
@@ -51,11 +54,15 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
+          proxyContract: "OptimizedTransparentUpgradeableProxy",
           execute: {
             methodName: "initialize",
             args: [comptrollerProxy.address, rewardTokenAddress, maxLoopsLimit, accessControlAddress],
           },
+          viaAdminContract: {
+            name: "DefaultProxyAdmin",
+            artifact: defaultProxyAdmin,
+          },
           upgradeIndex: 0,
```
Comment (chechu) [inline deploy/010-deploy-reward-distributors.ts:66]: Don't we need the `maxFeePerGas` property here for zkSync?

File: deploy/017-update-pool-registry.ts [hunk 5]
```diff
@@ -10,14 +10,20 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
   const { deployer } = await getNamedAccounts();
   const { preconfiguredAddresses } = await getConfig(hre.network.name);
   const proxyOwnerAddress = await toAddress(preconfiguredAddresses.NormalTimelock || "account:deployer", hre);
-
+  const defaultProxyAdmin = await hre.artifacts.readArtifact(
+    "hardhat-deploy/solc_0.8/openzeppelin/proxy/transparent/ProxyAdmin.sol:ProxyAdmin",
+  );
```
Comment (chechu) [inline deploy/017-update-pool-registry.ts:15]: Is this going to work in the `hardhat` network?

File: helpers/deploymentConfig.ts [hunk 6]
```diff
@@ -4308,6 +4319,159 @@ export const globalConfig: NetworkConfig = {
...
+        isMock: false,
+        name: "Wrapped Ether",
+        symbol: "WETH",
+        decimals: 18,
+        tokenAddress: "0x53F7e72C7ac55b44c7cd73cC13D4EF4b121678e6",
+      },
+      {
+        isMock: true,
+        name: "USD Coin",
+        symbol: "USDC",
```
Comment (chechu) [inline helpers/deploymentConfig.ts:4341]: `USDC.e`?

File: helpers/deploymentConfig.ts [hunk 7]
```diff
@@ -4308,6 +4319,159 @@ export const globalConfig: NetworkConfig = {
...
+            liquidationThreshold: convertToUnit("0.8", 18),
+            reserveFactor: convertToUnit("0.1", 18),
+            initialSupply: convertToUnit("5000", 6), // 5,000 USDT
+            supplyCap: convertToUnit(20_000_000, 6),
+            borrowCap: convertToUnit(18_000_000, 6),
+            reduceReservesBlockDelta: REDUCE_RESERVES_BLOCK_DELTA_ZKSYNC_SEPOLIA,
+            vTokenReceiver: preconfiguredAddresses.zksyncsepolia.VTreasury,
+          },
+          {
+            name: "Venus USDC (Core)",
```
Comment (chechu) [inline helpers/deploymentConfig.ts:4430]: `Venus USDC.e (Core)`?

File: deploy/017-update-pool-registry.ts (hunk 5 above)
Comment (GitGuru7) [reply deploy/017-update-pool-registry.ts:15]: Yes, it will

File: helpers/deploymentConfig.ts [hunk 8]
```diff
@@ -4308,6 +4319,159 @@ export const globalConfig: NetworkConfig = {
...
+      {
+        isMock: false,
+        name: "Wrapped Ether",
+        symbol: "WETH",
+        decimals: 18,
+        tokenAddress: "0x53F7e72C7ac55b44c7cd73cC13D4EF4b121678e6",
+      },
+      {
+        isMock: true,
+        name: "USD Coin",
```
[the remark this answers is not in the document]
Comment (GitGuru7) [reply helpers/deploymentConfig.ts]: [97ceecb](https://github.com/VenusProtocol/isolated-pools/pull/423/commits/97ceecbacd5e991f8703fa6cd39bae6e252742dd)

File: helpers/deploymentConfig.ts (hunk 6 above)
Comment (GitGuru7) [reply helpers/deploymentConfig.ts]: [97ceecb](https://github.com/VenusProtocol/isolated-pools/pull/423/commits/97ceecbacd5e991f8703fa6cd39bae6e252742dd)

File: helpers/deploymentConfig.ts (hunk 7 above)
Comment (GitGuru7) [reply helpers/deploymentConfig.ts]: [97ceecb](https://github.com/VenusProtocol/isolated-pools/pull/423/commits/97ceecbacd5e991f8703fa6cd39bae6e252742dd)

File: deploy/006-deploy-pool-registry.ts (hunk 2 above)
Comment (GitGuru7) [reply deploy/006-deploy-pool-registry.ts:35]: Yes, it will be

File: deploy/006-deploy-pool-registry.ts (hunk 1 above)
Comment (GitGuru7) [reply deploy/006-deploy-pool-registry.ts]: We need it occasionally. It is due to incorrect fee calculation on zksync sepolia. So, I am removing it as of now.

File: deploy/010-deploy-reward-distributors.ts (hunk 4 above)
Comment (GitGuru7) [reply deploy/010-deploy-reward-distributors.ts:66]: We need it occasionally. It is due to incorrect fee calculation on zksync sepolia.

File: deploy/006-deploy-pool-registry.ts (hunk 1 above)
Comment (GitGuru7) [reply deploy/006-deploy-pool-registry.ts]: [2837377](https://github.com/VenusProtocol/isolated-pools/pull/423/commits/2837377e9893ca5b18fccf6116314e3fd8f81478)

File: deploy/006-deploy-pool-registry.ts (hunk 3 above)
Comment (GitGuru7) [reply deploy/006-deploy-pool-registry.ts:24]: [2837377](https://github.com/VenusProtocol/isolated-pools/pull/423/commits/2837377e9893ca5b18fccf6116314e3fd8f81478)

**Review by chechu — APPROVED** [review]

**Pull request merged**